frontend/appflowy_flutter/lib/plugins/document/presentation/share/share_button.dart

```diff
@@ -1,39 +1,36 @@
-library document_plugin;
-
 import 'package:appflowy/generated/locale_keys.g.dart';
 import 'package:appflowy/startup/startup.dart';
 import 'package:appflowy/plugins/document/application/share_bloc.dart';
+import 'package:appflowy/util/file_picker/file_picker_service.dart';
 import 'package:appflowy/workspace/presentation/home/toast.dart';
 import 'package:appflowy/workspace/presentation/widgets/pop_up_action.dart';
 import 'package:appflowy_backend/protobuf/flowy-document2/entities.pb.dart';
 import 'package:appflowy_popover/appflowy_popover.dart';
-import 'package:clipboard/clipboard.dart';
 import 'package:easy_localization/easy_localization.dart';
-import 'package:file_picker/file_picker.dart';
 import 'package:flowy_infra_ui/widget/rounded_button.dart';
-import 'package:appflowy_backend/log.dart';
 import 'package:appflowy_backend/protobuf/flowy-error/errors.pb.dart';
 import 'package:appflowy_backend/protobuf/flowy-folder2/view.pb.dart';
 import 'package:flutter/material.dart';
 import 'package:flutter_bloc/flutter_bloc.dart';
 
 class DocumentShareButton extends StatelessWidget {
+  const DocumentShareButton({
+    super.key,
+    required this.view,
+  });
+
   final ViewPB view;
-  DocumentShareButton({Key? key, required this.view})
-      : super(key: ValueKey(view.hashCode));
 
   @override
   Widget build(BuildContext context) {
     return BlocProvider(
       create: (context) => getIt<DocShareBloc>(param1: view),
       child: BlocListener<DocShareBloc, DocShareState>(
         listener: (context, state) {
-          state.map(
-            initial: (_) {},
-            loading: (_) {},
+          state.mapOrNull(
             finish: (state) {
               state.successOrFail.fold(
-                _handleExportData,
+                (data) => _handleExportData(context, data),
```

Comment:
```suggestion
                (data) {
                  if (data.exportType == ExportType.Markdown) {
                    showSnackBarMessage(
                      context,
                      LocaleKeys.settings_files_exportFileSuccess.tr(),
                    );
                  }
                },
```
Then you can remove `_handleExportData`. If we don't intend to use it at some point at least.

Another option would be to add a snackbar for the two other cases as well. (See below)

Reply:
I believe it is better to keep the `_handleExportData` function. The other export logic will be implemented in the future. I don't want the `build` function to become too long.